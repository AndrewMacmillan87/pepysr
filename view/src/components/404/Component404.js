import React from "react";
import { Link } from "react-router-dom";

import "./component404.css";

/**
 * @description    React functional component for rendering
 *                 a message to the user in the event that
 *                 they attempt to navigate to a page that
 *                 doesn't exist
 * @return         React JSX element containing the
 *                 content to be displayed
 */
const Component404 = () => {
    return (
        <section>
            <div className='container-404'>
                <p className='msg-404'>
                    404: Oops, we couldn't find that page :(
                </p>
                <Link to='/'>
                    <button className='button' type='button'>
                        Back
                    </button>
                </Link>
            </div>
        </section>
    );
};

export default Component404;
