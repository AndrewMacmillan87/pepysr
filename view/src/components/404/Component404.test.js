import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

// Redux
import { Provider } from "react-redux";
import { reduxStore } from "../../redux/store/reduxStore";

import Component404 from "./Component404";

test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <Provider store={reduxStore}>
            <Router>
                <Component404 />
            </Router>
        </Provider>,
        div
    );
    ReactDOM.unmountComponentAtNode(div);
});

it("renders correctly", () => {
    const tree = renderer
        .create(
            <Provider store={reduxStore}>
                <Router>
                    <Component404 />
                </Router>
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
