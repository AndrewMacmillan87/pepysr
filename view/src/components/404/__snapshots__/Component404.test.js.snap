// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <div
    className="container-404"
  >
    <p
      className="msg-404"
    >
      404: Oops, we couldn't find that page :(
    </p>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="button"
        type="button"
      >
        Back
      </button>
    </a>
  </div>
</section>
`;
